apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/exception/Exceptions.kt

```diff
@@ -98,11 +98,24 @@ class CacheMissException(val key: String, val fieldName: String? = null) : Apoll
 class HttpCacheMissException(message: String, cause: Exception? = null) : ApolloException(message = message, cause = cause)
 
 /**
- * Multiple exceptions happened, for an exemple with a [CacheFirst] fetch policy
+ * Multiple exceptions happened, for an example with a [CacheFirst] fetch policy
  */
-class ApolloCompositeException(first: Throwable?, second: Throwable?) : ApolloException(message = "multiple exceptions happened", second) {
-  val first = (first as? ApolloException) ?: throw RuntimeException("unexpected first exception", first)
-  val second = (second as? ApolloException) ?: throw RuntimeException("unexpected second exception", second)
+class ApolloCompositeException(
+    private val original: Throwable?, private val suppressed: Throwable?
+) : ApolloException(message = "multiple exceptions happened", original ?: suppressed) {
```

Comment:
What's the reason to put the first exception as the cause when it was the second before? I'm not sure there are strong reasons to prefer one over the other but if we change it we should document it.

Reply:
CacheFirstInterceptor work is to fetch from cache first then network, if cacheMightThrow throws an exception it should be logged first and networkMightThrow should suppressed by it and same for NetworkFirstInterceptor and CacheAndNetworkInterceptor
This is the reason why first operations exception is thrown first and second is suppressed

Reply:
For the cause, I don't have a strong opinion either - in fact I'm not even sure it makes sense to set the cause at all 😅: `ApolloCompositeException` means "2 exception happened", is it really true to say that one is the _cause_ of the other? But I kind of agree that changing this now could be a bit dangerous if existing code depends on it.

For the suppressed... Well now that I think even more about it, I think both should be marked as suppressed? There can be more than 1, right?

Reply:
Yes, it can be multiple as it supports list of suppressed exception

Reply:
So one doubt here if we keep both as suppressed, default cause will be null
````
ApolloException(message = "multiple exceptions happened", null)
````
Which will change the current behavior

Reply:
Agree, i also think that would be safe and we can update it in future

Reply:
Which one you think will do
````
second ?: first 
````
or 
````
second
````

Reply:
Since we're going for backward compatibility, let's keep

```
second
```

Reply:
Updated